From 33dabdc9143b9051578aa3b4d23b2512448ceb91 Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 26 Aug 2014 12:49:13 -0400
Subject: [tools] cpupower: fix wrong err msg not supported vs not available

Message-id: <1409057382-26432-3-git-send-email-prarit@redhat.com>
Patchwork-id: 89078
O-Subject: [RHEL7.1 PATCH BZ 1133893 02/31] tools: cpupower: fix wrong err msg not supported vs not available
Bugzilla: 1133893
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1133893

commit f4a5d17efec95ee74113df62943494a2197c8bae
Author: Thomas Renninger <trenn@suse.de>
Date:   Sat Nov 16 14:39:22 2013 +0100

    tools: cpupower: fix wrong err msg not supported vs not available

    idlestates in sysfs are counted from 0.

    This fixes a wrong error message.
    Current behavior on a machine with 4 sleep states is:

    cpupower idle-set -e 4
    Idlestate 4 enabled on CPU 0

    -----Wrong---------------------
    cpupower idle-set -e 5
    Idlestate enabling not supported by kernel
    -----Must and now will be -----
    cpupower idle-set -e 5
    Idlestate 6 not available on CPU 0
    -------------------------------

    cpupower idle-set -e 6
    Idlestate 6 not available on CPU 0

    Signed-off-by: Thomas Renninger <trenn@suse.de>
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Cc: Tony Camuso <tcamuso@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/power/cpupower/utils/helpers/sysfs.c b/tools/power/cpupower/utils/helpers/sysfs.c
index 5cdc600..851c7a1 100644
--- a/tools/power/cpupower/utils/helpers/sysfs.c
+++ b/tools/power/cpupower/utils/helpers/sysfs.c
@@ -278,7 +278,7 @@ static char *sysfs_idlestate_get_one_string(unsigned int cpu,
 int sysfs_is_idlestate_disabled(unsigned int cpu,
     unsigned int idlestate)
 {
- if (sysfs_get_idlestate_count(cpu) < idlestate)
+ if (sysfs_get_idlestate_count(cpu) <= idlestate)
   return -1;
 
  if (!sysfs_idlestate_file_exists(cpu, idlestate,
@@ -303,7 +303,7 @@ int sysfs_idlestate_disable(unsigned int cpu,
  char value[SYSFS_PATH_MAX];
  int bytes_written;
 
- if (sysfs_get_idlestate_count(cpu) < idlestate)
+ if (sysfs_get_idlestate_count(cpu) <= idlestate)
   return -1;
 
  if (!sysfs_idlestate_file_exists(cpu, idlestate,
-- 
1.7.1